File: packages/contracts/src/v1/agent.ts
import { z } from "zod";
import { VersionSchema } from "./common";
import { SuggestionDiffSchema } from "./suggestions";

export const AgentContextBlockSchema = z.object({
  blockId: z.string().uuid(),
  content: z.string(),
  score: z.number().min(0).max(1).optional()
});

export const RewriteSectionRequestSchema = z.object({
  version: VersionSchema,
  documentId: z.string().uuid(),
  sectionId: z.string().uuid(),
  content: z.string().min(1),
  context: z.array(AgentContextBlockSchema),
  tone: z.string().default("neutral")
});

export const SummarizeDocumentRequestSchema = z.object({
  version: VersionSchema,
  documentId: z.string().uuid(),
  targetSectionIds: z.array(z.string().uuid()).min(1),
  context: z.array(AgentContextBlockSchema)
});

export const AgentSuggestionResponseSchema = z.object({
  version: VersionSchema,
  requestId: z.string().min(1),
  suggestion: SuggestionDiffSchema
});

export const AgentEnqueueJobResponseSchema = z.object({
  version: VersionSchema,
  taskId: z.string().min(1)
});

export type RewriteSectionRequest = z.infer<typeof RewriteSectionRequestSchema>;
export type SummarizeDocumentRequest = z.infer<typeof SummarizeDocumentRequestSchema>;

File: apps/agent/schemas/agent.py
from pydantic import BaseModel, ConfigDict, Field


class AgentContextBlock(BaseModel):
    model_config = ConfigDict(extra="forbid")

    blockId: str
    content: str
    score: float | None = Field(default=None, ge=0, le=1)


class RewriteSectionRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    version: str = "v1"
    documentId: str
    sectionId: str
    content: str
    context: list[AgentContextBlock]
    tone: str = "neutral"


class SummarizeDocumentRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    version: str = "v1"
    documentId: str
    targetSectionIds: list[str]
    context: list[AgentContextBlock]


class DiffOperation(BaseModel):
    model_config = ConfigDict(extra="forbid")

    operation: str
    targetBlockId: str
    content: str | None = None
    startOffset: int | None = None
    endOffset: int | None = None


class SuggestionDiff(BaseModel):
    model_config = ConfigDict(extra="forbid")

    version: str = "v1"
    suggestionId: str
    documentId: str
    baseSnapshotVersion: int
    operations: list[DiffOperation]
    reason: str


class AgentSuggestionResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    version: str = "v1"
    requestId: str
    suggestion: SuggestionDiff
